insights/parsers/messages.py

```diff
@@ -86,3 +86,23 @@ def get(self, s):
                     msg_info['procname'] = info_splits[4]
                 r.append(msg_info)
         return r
+
+    def get_after(self, timestamp, lines=None):
+        """
+        Get a list of lines after the given time stamp.
+
+        Parameters:
+            timestamp(datetime.datetime): log lines after this time are
+                returned.
+            lines(list): the list of log lines to search (e.g. from a get).
+                If not supplied, all available lines are searched.
+
+        Returns:
+            (list): The list of log lines with time stamps after the given
+                date and time.
+
+        *NOTE*: Because /var/log/messages timestamps by default have no year,
+        you must supply the year in your timestamp as 1900.  This may also
+        cause confusion around the end of the year.
```

Comment:
I'm really not sure how.

The main problem is that if we see a skip back in time - let's say over 11 months - it indicates that all the logs read up until now are from the previous year.  But in the current implementation we've yielded those logs, so we can't go back and re-evaluate those logs because we now know that they're in the previous year.

I'm trying to avoid just focusing on the specific use-case of finding all the logs after the last collection run.  There are other dates, such as the date the configuration file was changed, that a rule could use in get_after() that may be longer than 24 hours since the last update, and may be earlier than that roll-over from December to January.

One possible approach might be to use the year given in the timestamp argument itself as the assumed year for logs without a year.  When the date is read from the log line, we check whether the log date is greater than 11 months ahead of the sought timestamp.  If it is, this indicates that the log line is from December and the sought date is in January, so we shift that log date back a year to put it in the previous year relative to the sought date.  That's done for every log line, so even if for some reason (e.g. Apache log lines) the logs go backward in time, log dates are always corrected as above.

I'm not super keen on adding this kind of specialised processing into what is already a large and complicated function, but I'll give it a go and see how it turns out.